custom-routes-vis/src/main/webapp/script.js

```diff
@@ -65,7 +68,7 @@ function createMarker(containerId, label, title, latLng) {
 
   marker.addListener('dragend', function(event) {
     updateCoordinates(event.latLng.lat(), event.latLng.lng(), containerId);
-    if (originDestinationMarkers.length == 2 
+    if (originDestinationMarkers.length === 2 
         && originDestinationMarkers[0].getVisible() 
         && originDestinationMarkers[1].getVisible()) {
```

Comment:
As I mentioned, in another comment I don't think you need to check that there are 2 markers. The markers are both created to not be visible, one after the other. So if they are visible, then they have both been created.

Reply:
I need to check because otherwise if you drag the first marker when the second has not been created, it will give you an index out of bound error

Reply:
You should not be able to even move the first marker before the second is created...they are created on after the other, and before they are even visible? The user would need to drag the invisible marker within the micro seconds before the second marker is made.

Reply:
lmao yeah but still it wouldn't hurt to check